winml/test/api/LearningModelSessionAPITest.cpp

```diff
@@ -311,6 +311,37 @@ static void EvaluateSessionAndCloseModel()
     WINML_EXPECT_NO_THROW(::EvaluateSessionAndCloseModelHelper(LearningModelDeviceKind::Cpu, false));
 }
 
+static void NamedDimensionOverride() 
+{
+  LearningModel model = nullptr;
+  WINML_EXPECT_NO_THROW(APITest::LoadModel(L"fns-candy.onnx", model));
+
+  LearningModelDevice device(nullptr);
+  WINML_EXPECT_NO_THROW(device = LearningModelDevice(LearningModelDeviceKind::Cpu));
+
+  LearningModelSessionOptions options;
+  options.OverrideNamedDimension(L"None", 5);
+  
+  // Verifies that if a Dim name doesn't exist the named dimension override does nothing
```

Comment:
should it do nothing? or should it be an error?

Reply:
Right, is there supposed to be an HRESULT returned indicating that the dimension name is incorrect? 

---
In reply to: [459783063](https://github.com/microsoft/onnxruntime/pull/4606#discussion_r459783063) [](ancestors = 459783063)

Reply:
The ORT Graph transformer will look for matches of existing inputs and ignore all other overrides that are specified. So to implement anything else on the WinML projection would be breaking parity with ORT and I don't think it really adds value. I added this test to just verify that this ORT behavior stays constant.